core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/report/ReportEditViewModel.kt

```diff
@@ -0,0 +1,230 @@
+package com.ustadmobile.core.viewmodel.report
+
+import com.ustadmobile.core.MR
+import com.ustadmobile.core.domain.report.model.ReportFilter3
+import com.ustadmobile.core.domain.report.model.ReportOptions2
+import com.ustadmobile.core.domain.report.model.ReportSeries2
+import com.ustadmobile.core.impl.appstate.ActionBarButtonUiState
+import com.ustadmobile.core.impl.appstate.AppUiState
+import com.ustadmobile.core.impl.appstate.LoadingUiState
+import com.ustadmobile.core.impl.nav.UstadSavedStateHandle
+import com.ustadmobile.core.util.ext.replace
+import com.ustadmobile.core.view.UstadView
+import com.ustadmobile.core.viewmodel.ReportFilterEditViewModel
+import com.ustadmobile.core.viewmodel.UstadEditViewModel
+import com.ustadmobile.door.ext.withDoorTransactionAsync
+import com.ustadmobile.lib.db.entities.Report
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.kodein.di.DI
+
+class ReportEditViewModel(
+    di: DI,
+    savedStateHandle: UstadSavedStateHandle,
+    destName: String = ""
+) : UstadEditViewModel(di, savedStateHandle, DEST_NAME) {
+
+    private val _uiState: MutableStateFlow<ReportEditUiState> =
+        MutableStateFlow(ReportEditUiState())
+    val uiState: Flow<ReportEditUiState> = _uiState.asStateFlow()
+    private val entityUid: Long
```

Comment:
You don't need your own ```entityUid```. You can get this from the parent class's ```entityUidArg```. I think this was in PersonEditViewModel because it was created before entityUidArg was added to the parent class.